### DISCUSSION TITLE
How to solve optimal control problem with delay?

### DISCUSSION THREAD

**Discussion by Song921012** [infiniteopt/InfiniteOpt.jl#285] [Q&A]
Hello, can `Infiniteopt.jl` solve optimal control problem with delay? For example, 
![image](https://user-images.githubusercontent.com/61453779/187486490-ae6d431b-bbd1-4e11-aa38-cf8717ba0a73.png)

```math
\left\{\begin{array}{l}
\min\int_0^T e^{-\beta t}\left(C_E \frac{u(t)^3}{x(t)}-p u(t)\right) d t+0.1 T^2 \\
\dot{x}=a x(t)\left(1-\frac{x(t-h)}{b}\right)-u(t) \\
x(t)=2 \forall t \in[-h, 0] \quad, \quad x(t) \geq 2 \forall t \in[0, T] \\
u(t) \geq 0 \forall t \in[0, T] \\
T \text{ free. }
\end{array}\right.
```

**Comment by pulsipher** [comment]
InfiniteOpt does not directly support problems where the limits of an infinite parameter (e.g., time) are decision variables (i.e., T in this case). However, such problems can often be reformulated into standard dynamic optimization problems (with a fixed horizon) as demonstrated on some of the other discussions on the forum and as is common in optimal control literature. In this case, I suspect this can be reformulated in like manner to minimum time problems since I beleive the final time T is a variable and the initial time -h is fixed.
